libethereum/Block.cpp

```diff
@@ -136,47 +136,59 @@ void Block::noteChain(BlockChain const& _bc)
     }
 }
 
-PopulationStatistics Block::populateFromChain(BlockChain const& _bc, h256 const& _h, ImportRequirements::value _ir)
+void Block::populateFromChain(BlockChain const& _bc, h256 const& _h)
 {
-    noteChain(_bc);
-
-    PopulationStatistics ret { 0.0, 0.0 };
-
     if (!_bc.isKnown(_h))
     {
-        // Might be worth throwing here.
-        cwarn << "Invalid block given for state population: " << _h;
+        cerr << "Invalid block given for state population: " << _h;
         BOOST_THROW_EXCEPTION(BlockNotFound() << errinfo_target(_h));
     }
 
-    auto b = _bc.block(_h);
-    BlockHeader bi(b);		// No need to check - it's already in the DB.
-    if (bi.number())
+    auto const& blockBytes = _bc.block(_h);
+    m_currentBytes = blockBytes;
```

Comment:
Did it fill `m_currentBytes` before?

Reply:
No it didn’t, and I think as a side effect we return 0 size value to eth_getBlock RPC function (see this issue: https://github.com/ethereum/aleth/issues/5797)

I think conceptually it makes sense to set the bytes here since they are the backing data for the block, though in practice I don’t know if they will be useful to callers.

Reply:
Not sure if it makes sense. 

I think the idea here was that the binary representation is useful only in case we're going to try to seal this block (`isSealed` implementation points to this). So for RPC methods it shouldn't be needed and we only waste memory and time copying it.

Reply:
regarding size in RPC, see my comment in the issue